lib/compute/agent-node-config.ts

```diff
@@ -23,7 +23,7 @@ export interface AgentNodeProps {
   amiId: string;
   instanceType: string;
   customDeviceMapping: string;
-  workerLabelString: string;
+  workerLabelString: string[];
```

Comment:
How about make it `[string, DeploymentType]`. This way it will by default only accept the specified enum values.

Reply:
We already have a check for the enum values
> 
> else if (!Object.values(DeploymentType).includes(jenkinsInstanceType as DeploymentType)) {
>       throw new Error(`Invalid jenkinsInstanceType value: ${jenkinsInstanceType}`);

and we have hardcoded the DeploymentType for each agent property anyway

> workerLabelString: ['Jenkins-Agent-AL2023-X64-C54xlarge-Single-Host', 'BTR'],  

Do we still require to change the type @gaiksaya?

Reply:
Addressed all other comments in the latest commit @gaiksaya

Reply:
The condition will only be checked if the given value is undefined as per this code: https://github.com/opensearch-project/opensearch-ci/pull/513/files#diff-4b6fe4122d26e45b42b65d74533746571fed0807fe76bb2df2ea33b75227e004R130-R134
We want to check despite of that

Reply:
Synced up offline. The if else block does the needful. We can iterate on this logic later.